DSCResources/MSFT_xCertificateImport/MSFT_xCertificateImport.psm1

```diff
@@ -0,0 +1,268 @@
+    }
+}
+
+<#
+.SYNOPSIS
+    Validates whether a given certificate is valid based on the hash algoritms available on the system.
+
+.PARAMETER Thumbprint
+    One or more thumbprints to validate.
+
+.PARAMETER Quiet
+    Returns $false if the thumbprint is not valid. By default this function throws an exception if validation fails.
+
+.EXAMPLE
+    Validate-Thumbprint fd94e3a5a7991cb6ed3cd5dd01045edf7e2284de
+
+.EXAMPLE
+    Validate-Thumbprint fd94e3a5a7991cb6ed3cd5dd01045edf7e2284de,0000e3a5a7991cb6ed3cd5dd01045edf7e220000 -Quiet
+
+.EXAMPLE
+    gci Cert:\LocalMachine -Recurse | ? { $_.Thumbprint } | select -exp Thumbprint | Validate-Thumbprint -Verbose
+#>
+function Validate-Thumbprint 
+{
+    [CmdletBinding()]
+    param(
+        [Parameter(
+            Mandatory,
+            ValueFromPipeline
+        )]
+        [ValidateNotNullOrEmpty()]
+        [String[]]
+        $Thumbprint ,
+
+        [Parameter()]
+        [Switch]
+        $Quiet
+    )
+
+    Begin 
+    {
+        $validHashes = [System.AppDomain]::CurrentDomain.GetAssemblies().GetTypes() | Where-Object {
+            $_.BaseType.BaseType -eq [System.Security.Cryptography.HashAlgorithm] -and
+            $_.Name -cmatch 'Managed$'
+        } | ForEach-Object {
+            New-Object PSObject -Property @{
+                Hash = $_.BaseType.Name
+                BitSize = (New-Object $_).HashSize
+            } | Add-Member -MemberType ScriptProperty -Name HexLength -Value { 
+                $this.BitSize / 4 
+            } -PassThru
+        }
+    }
+
+    Process 
+    {
+        foreach($hash in $Thumbprint) 
+        {
+            $isValid = $false
+
+            foreach($algorithm in $validHashes)
+            {
+                if ($hash -cmatch "^[a-fA-F0-9]{$($algorithm.HexLength)}$")
+                {
+                    Write-Verbose -Message "'$hash' is a valid $($algorithm.Hash) hash."
+                    $isValid = $true
+                }
+            }
+            
+            if ($Quiet -or $isValid)
+            {
+                $isValid
+            }
+            else
+            {
+                throw [System.ArgumentException]"'$hash' is not a valid hash."
+            }
+        }
+    }
+}
+
+function Get-TargetResource 
+{
+    [CmdletBinding()]
+    [OutputType([Hashtable])]
+    param(
+        [Parameter(
+            Mandatory
+        )]
+        [ValidateScript( {
+            $_ | Validate-Thumbprint
+        } )]
+        [String]
+        $Thumbprint ,
+
+        [Parameter(
+            Mandatory
+        )]
+        [ValidateScript( {
+            $_ | Validate-CertificatePath
+        } )]
+        [String]
+        $Path ,
+
+        [Parameter()]
+        [ValidateSet(
+             'LocalMachine'
+        )]
+        [String]
+        $Location = 'LocalMachine' ,
+
+        [Parameter()]
+        [ValidateNotNullOrEmpty()]
+        [String]
+        $Store = 'My'
+    )
+
+    @{
+        Thumbprint = $Thumbprint
```

Comment:
Of course it should. Again, the implementation was a copy of the xPfxImport-resource and it has the same implementation. I will fix it in this resource and add an issue for the xPfxImport-resource.